internal/dataplane/kong_client.go

```diff
@@ -409,19 +413,27 @@ func (c *KongClient) Update(ctx context.Context) error {
 		formatVersion = "3.0"
 	}
 
+	configStatus := new(ConfigStatus)
+	*configStatus = ConfigStatusOK
+	defer func(status *ConfigStatus) {
+		c.configStatusNotifier.NotifyConfigStatus(*status)
+	}(configStatus)
```

Comment:
Wouldn't this work the same without the allocation:

```suggestion
	configStatus := ConfigStatusOK
	defer func(status *ConfigStatus) {
		c.configStatusNotifier.NotifyConfigStatus(*status)
	}(&configStatus)
```

?

Reply:
On the other hand: I feel this is a bit too complicated for what it's doing. Why not just call

```
c.configStatusNotifier.NotifyConfigStatus(ConfigStatusOK)
```

or

```
c.configStatusNotifier.NotifyConfigStatus(ConfigStatusOK)
```

or

```
c.configStatusNotifier.NotifyConfigStatus(ConfigStatusApplyFailed)
```

Where we need it to be called. Less code, more readable: win-win. Or do we need to ensure that this is the last thing being called via a `defer`?

Reply:
removed the `defer` and directly do the notify. My intention to use `defer` is to run the notification anyway if  the update returns error prematurely.